### ISSUE TITLE
[Enhancement] Update OS packages for Harvester component images

### ISSUE THREAD

**Issue by guangbochen** [harvester/harvester#3159]
**Describe the bug**
We need to update OS packages for Harvester component images as they are either outdated or need a patch update. 

**Expected behavior**
Update OS packages for Harvester component images
- [x] https://github.com/harvester/harvester/pull/3186
- [x] https://github.com/harvester/node-manager - v0.1.3
- [x] https://github.com/harvester/network-controller-harvester - v0.3.1
- [x] https://github.com/harvester/node-disk-manager - v0.4.8
- [x] https://github.com/harvester/vm-import-controller - v0.1.2
- [x] https://github.com/harvester/pcidevices - v0.2.3
- [x] [Update OS](https://github.com/harvester/harvester-installer/pull/384)

**Environment**
 - Harvester ISO version: v1.1.1-rc1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): All

**Additional context**
N/A

**Comment by guangbochen** [comment]
**To Test:**
- Upgrade from v1.0.3 to the `master-head.iso` should work
    - pods in the `harvester-system` are all running properly
    - Kubevirt-related image is updated to `registry.suse.com/suse/sles/15.4$component:0.54.0-150400.3.7.1`
    - `vm-import` and `pci-devices` addon versions are updated to `v0.1.2` and `v0.2.3` respectively
    - basic VM features are working
    - OS kernel is updated to [5.3.18-150300.59.101.1 ](https://github.com/harvester/harvester-installer/pull/384)

- Upgrade from v1.1.0 to the `master-head.iso` should work
    - pods in the `harvester-system` are all running properly
    - Kubevirt-related image is updated to `registry.suse.com/suse/sles/15.4$component:0.54.0-150400.3.7.1`
    - `vm-import` and `pci-devices` addon versions are updated to `v0.1.2` and `v0.2.3` respectively
    - basic VM features are working
    - OS kernel is updated to [5.3.18-150300.59.101.1 ](https://github.com/harvester/harvester-installer/pull/384)

**Comment by TachunLin** [comment]
### (PASS) Upgrade from v1.0.3 to the master-head 

- [x] Upgrade from v1.0.3 to the `master-head.iso` can complete
![image](https://user-images.githubusercontent.com/29251855/204260475-cdef0bd6-dde0-4ec7-88c0-f88ef4b3fb80.png)

- [x] pods in the `harvester-system` are all running properly
- [x] Kubevirt-related image is updated to `registry.suse.com/suse/sles/15.4$component:0.54.0-150400.3.7.1`
  ```
  harvester-tmpgd:~ # kubectl get deployments -A -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.template.spec.containers[*].image}{"\n"}{end}'|column -t
  fleet-agent                            rancher/fleet-agent:v0.4.0                                                                                     
  fleet-controller                       rancher/fleet:v0.4.0                                                                                           
  gitjob                                 rancher/gitjob:v0.1.32                                                                                         
  rancher-logging                        rancher/mirrored-banzaicloud-logging-operator:3.17.7                                                           
  rancher-monitoring-grafana             rancher/mirrored-kiwigrid-k8s-sidecar:1.12.3                          rancher/mirrored-grafana-grafana:7.5.11  rancher/mirrored-library-nginx:1.21.1-alpine
  rancher-monitoring-kube-state-metrics  rancher/mirrored-kube-state-metrics-kube-state-metrics:v2.2.0                                                  
  rancher-monitoring-operator            rancher/mirrored-prometheus-operator-prometheus-operator:v0.50.0                                               
  rancher-monitoring-prometheus-adapter  rancher/mirrored-prometheus-adapter-prometheus-adapter:v0.9.0                                                  
  harvester-cluster-repo                 rancher/harvester-cluster-repo:master                                                                          
  rancher                                rancher/rancher:v2.6.9                                                                                         
  rancher-webhook                        rancher/rancher-webhook:v0.2.7                                                                                 
  system-upgrade-controller              rancher/system-upgrade-controller:v0.9.1                                                                       
  harvester                              rancher/harvester:master-head                                                                                  
  harvester-load-balancer                rancher/harvester-load-balancer:master-head                                                                    
  harvester-network-controller-manager   rancher/harvester-network-controller:master-head                                                               
  harvester-network-webhook              rancher/harvester-network-webhook:master-head                                                                  
  harvester-webhook                      rancher/harvester-webhook:master-head                                                                          
  virt-api                               registry.suse.com/suse/sles/15.4/virt-api:0.54.0-150400.3.7.1                                                  
  virt-controller                        registry.suse.com/suse/sles/15.4/virt-controller:0.54.0-150400.3.7.1                                           
  virt-operator                          registry.suse.com/suse/sles/15.4/virt-operator:0.54.0-150400.3.7.1                                             
  rke2-coredns-rke2-coredns              rancher/hardened-coredns:v1.9.3-build20221011                                                                  
  rke2-coredns-rke2-coredns-autoscaler   rancher/hardened-cluster-autoscaler:v1.8.5-build20221011                                                       
  rke2-metrics-server                    rancher/hardened-k8s-metrics-server:v0.6.1-build20221011                                                       
  snapshot-controller                    k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0                                                              
  csi-attacher                           longhornio/csi-attacher:v3.4.0                                                                                 
  csi-provisioner                        longhornio/csi-provisioner:v2.1.2                                                                              
  csi-resizer                            longhornio/csi-resizer:v1.2.0                                                                                  
  csi-snapshotter                        longhornio/csi-snapshotter:v3.0.3                                                                              
  longhorn-admission-webhook             longhornio/longhorn-manager:v1.3.2                                                                             
  longhorn-conversion-webhook            longhornio/longhorn-manager:v1.3.2                                                                             
  longhorn-driver-deployer               longhornio/longhorn-manager:v1.3.2                                                                             
  longhorn-ui                            longhornio/longhorn-ui:v1.3.2
  ```
- [x] `vm-import` are updated to `v0.1.2` respectively
![image](https://user-images.githubusercontent.com/29251855/204262248-2209e2a0-8367-46e1-bae8-8e18404acba0.png)


- [x] `pci-devices` addon versions are updated to `v0.2.3` respectively
![image](https://user-images.githubusercontent.com/29251855/204262199-a0c2df54-71d9-4a47-8ead-6d480f5af136.png)



- [x] Basic VM features are working
  - [x] Create, Delete VM
  - [x] Start, Stop, Restart VM
  - [x] Add volume
  - [x] Clone VM
  - [x] Migrate VM 

- [x] OS kernel is updated to [5.3.18-150300.59.101.1 ](https://github.com/harvester/harvester-installer/pull/384)
  ```
  harvester-tmpgd:~ # rpm -qa |grep grub
  grub2-2.04-150300.22.25.1.x86_64
  grub2-i386-pc-2.04-150300.22.25.1.noarch
  grub2-x86_64-efi-2.04-150300.22.25.1.noarch
  ```

**Comment by TachunLin** [comment]
### (PASS) Upgrade from v1.1.0 to the master-head 

- [x] Upgrade from v1.1.0 to the `master-head.iso` can complete
![image](https://user-images.githubusercontent.com/29251855/204271065-e1406f9f-e444-4557-ad2a-2819076095ea.png)

- [x] pods in the `harvester-system` are all running properly


- [x] Kubevirt-related image is updated to `registry.suse.com/suse/sles/15.4$component:0.54.0-150400.3.7.1`
  ```
  harvester-node-1:~ # kubectl get deployments -A -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.template.spec.containers[*].image}{"\n"}{end}'|column -t
  fleet-agent                            rancher/fleet-agent:v0.4.0                                                                                     
  fleet-controller                       rancher/fleet:v0.4.0                                                                                           
  gitjob                                 rancher/gitjob:v0.1.32                                                                                         
  rancher-logging                        rancher/mirrored-banzaicloud-logging-operator:3.17.7                                                           
  rancher-monitoring-grafana             rancher/mirrored-kiwigrid-k8s-sidecar:1.12.3                          rancher/mirrored-grafana-grafana:7.5.11  rancher/mirrored-library-nginx:1.21.1-alpine
  rancher-monitoring-kube-state-metrics  rancher/mirrored-kube-state-metrics-kube-state-metrics:v2.2.0                                                  
  rancher-monitoring-operator            rancher/mirrored-prometheus-operator-prometheus-operator:v0.50.0                                               
  rancher-monitoring-prometheus-adapter  rancher/mirrored-prometheus-adapter-prometheus-adapter:v0.9.0                                                  
  harvester-cluster-repo                 rancher/harvester-cluster-repo:master                                                                          
  rancher                                rancher/rancher:v2.6.9                                                                                         
  rancher-webhook                        rancher/rancher-webhook:v0.2.7                                                                                 
  system-upgrade-controller              rancher/system-upgrade-controller:v0.9.1                                                                       
  harvester                              rancher/harvester:master-head                                                                                  
  harvester-load-balancer                rancher/harvester-load-balancer:master-head                                                                    
  harvester-network-controller-manager   rancher/harvester-network-controller:master-head                                                               
  harvester-network-webhook              rancher/harvester-network-webhook:master-head                                                                  
  harvester-webhook                      rancher/harvester-webhook:master-head                                                                          
  virt-api                               registry.suse.com/suse/sles/15.4/virt-api:0.54.0-150400.3.7.1                                                  
  virt-controller                        registry.suse.com/suse/sles/15.4/virt-controller:0.54.0-150400.3.7.1                                           
  virt-operator                          registry.suse.com/suse/sles/15.4/virt-operator:0.54.0-150400.3.7.1                                             
  rke2-coredns-rke2-coredns              rancher/hardened-coredns:v1.9.3-build20221011                                                                  
  rke2-coredns-rke2-coredns-autoscaler   rancher/hardened-cluster-autoscaler:v1.8.5-build20221011                                                       
  rke2-metrics-server                    rancher/hardened-k8s-metrics-server:v0.6.1-build20221011                                                       
  snapshot-controller                    k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0                                                              
  csi-attacher                           longhornio/csi-attacher:v3.4.0                                                                                 
  csi-provisioner                        longhornio/csi-provisioner:v2.1.2                                                                              
  csi-resizer                            longhornio/csi-resizer:v1.2.0                                                                                  
  csi-snapshotter                        longhornio/csi-snapshotter:v3.0.3                                                                              
  longhorn-admission-webhook             longhornio/longhorn-manager:v1.3.2                                                                             
  longhorn-conversion-webhook            longhornio/longhorn-manager:v1.3.2                                                                             
  longhorn-driver-deployer               longhornio/longhorn-manager:v1.3.2                                                                             
  longhorn-ui                            longhornio/longhorn-ui:v1.3.2
  ```

  
- [x] `vm-import` are updated to `v0.1.2` respectively
  ![image](https://user-images.githubusercontent.com/29251855/204271615-8b3e3e6e-7ccf-4490-9c82-47987045318f.png)

- [x] `pci-devices` addon versions are updated to `v0.2.3` respectively
  ![image](https://user-images.githubusercontent.com/29251855/204271525-b5e99f07-5331-408e-bb69-5cdb4b3293dc.png)


- [x] Basic VM features are working
  - [x] Create, Delete VM
  - [x] Start, Stop, Restart VM
  - [x] Add volume
  - [x] Clone VM
  - [x] Migrate VM 

- [x] OS kernel is updated to [5.3.18-150300.59.101.1 ](https://github.com/harvester/harvester-installer/pull/384)
  ```
  harvester-node-1:~ # rpm -qa |grep grub
  grub2-2.04-150300.22.25.1.x86_64
  grub2-i386-pc-2.04-150300.22.25.1.noarch
  grub2-x86_64-efi-2.04-150300.22.25.1.noarch
  ```

**Comment by TachunLin** [comment]
Verified **fixed** on `master-0cdb5a5b-head` (11/28). Close this issue.

### Result
Verified can pass in the following upgrade path
1. Upgrade from v1.0.3 to v1.1.1 master-head [result](https://github.com/harvester/harvester/issues/3159#issuecomment-1328877606) 
2. Upgrade from v1.1.0 to v1.1.1 master-head [result](https://github.com/harvester/harvester/issues/3159#issuecomment-1328877606) 

### Test Information
* Test Environment: 4 nodes bare machines and kvm machines
* Harvester version: `master-0cdb5a5b-head` (11/28)

### Verify Steps
1. Prepare 4 nodes v1.0.3 or v1.1.0 Harvester cluster
2. Prepare some image and create some VMs
3. Upgrade to v1.1.1 master head release
4. Check can upgrade complete and system up
5. ssh to nodes
6. Execute `kubectl get pods -A` 
7. Check all images of deployments
    ```
    kubectl get deployments -A -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.template.spec.containers[*].image}{"\n"}{end}'|column -t
    ```
8. Edit the yaml of `vm-import` and `pci-devices` addon
9. Execute `rpm -qa |grep grub`
10. Run VM basic operations
